### PR TITLE
[NFC] Move more logic about unfuzzable tests to a shared location

### PR DESCRIPTION
**Pull request by kripken**
It turns out that #7165 was not enough because we had a second (!?)
list of tests to ignore, and also a condition. Move all that to the shared
location as well, continuing that PR.

Also remove `simd.wast` from the list, as that issue has been fixed.

### CODE REVIEW

File: scripts/test/fuzzing.py [hunk 1]
```diff
@@ -73,4 +75,22 @@
     'typed_continuations_contnew.wast',
     'typed_continuations_contbind.wast',
     'typed_continuations_suspend.wast',
+    # contains too many segments to run in a wasm VM
+    'limit-segments_disable-bulk-memory.wast',
+    # https://github.com/WebAssembly/binaryen/issues/3203
+    'simd.wast',
+    # corner cases of escaping of names is not interesting
+    'names.wast',
```
Comment (tlively) [inline scripts/test/fuzzing.py:83]: It's worth checking whether these two can be re-enabled now.

**Review by tlively — APPROVED** [review]
Nice simplification 👍

File: scripts/test/fuzzing.py (hunk 1 above)
Comment (kripken) [reply scripts/test/fuzzing.py:83]: Looks like `names.wast` causes errors,
```
$ wasm-metadce a.wat -o a.wat --graph-file a.json -all
wasm-metadce: /usr/local/google/home/azakai/Dev/binaryen/src/support/json.h:282: char* json::Value::parse(char*): Assertion `*curr == ','' failed.
Aborted
```
`a.wat` can be an empty wasm module. The [json file](https://gist.github.com/kripken/91055cca5b5578061f2527a07abcc0db) was created from names in `names.wast`, and I guess we are not escaping something there, or `json.h` isn't reading escaped names properly.

Comment (kripken) [reply scripts/test/fuzzing.py:83]: That json looks valid, or at least JS engines don't error on it. So `json.h` parsing might be the problem.

Comment (tlively) [reply scripts/test/fuzzing.py:83]: Probably needs to be updated to handle arbitrary unicode and possibly null characters in the middle of a string.

Comment (kripken) [reply scripts/test/fuzzing.py]: `simd.wast` otoh looks fine, I enabled it.
